python/utils/run_benchmarks.py

```diff
@@ -1,82 +1,85 @@
 import argparse
 import os
+from os import path
 import sys
-import time
 from collections import OrderedDict
 
 from scipy.io import mmread
 
 import cugraph
 import cudf
 
+from benchmark import (
+    Benchmark, logExeTime, printLastResult, noStdoutWrapper, nop
+)
+
+from asv_report import (
+    cugraph_update_asv
+)
 
 ########################################
 # Update this function to add new algos
 ########################################
-def getAlgoData(G, edgelist_gdf, args):
-    """
-    keyname = algo method/function name
-    args = args to pass the method/function (default is no args)
-    obj = object containing the method/function (default is the cugraph module)
-    extraWrappers = list of functions that return a callable, used for
-                    "wrapping" the algo to modify its environment, add timers,
-                    log calls, etc.
-    """
-    algoData = {"pagerank":
-                {"args": (G, args.damping_factor, None, args.max_iter,
-                          args.tolerance),
-                 },
-                "bfs":
-                {"args": (G, args.source, True),
-                 },
-                "sssp":
-                {"args": (G, args.source),
-                 "extraWrappers": [noStdoutWrapper],
-                 },
-                "jaccard":
-                {"args": (G,),
-                 },
-                "louvain":
-                {"args": (G,),
-                 },
-                "weakly_connected_components":
-                {"args": (G,),
-                 },
-                "overlap":
-                {"args": (G,),
-                 },
-                "triangles":
-                {"args": (G,),
-                 },
-                "spectralBalancedCutClustering":
-                {"args": (G, 2),
-                 },
-                "spectralModularityMaximizationClustering":
-                {"args": (G, 2),
-                 },
-                "renumber":
-                {"args": (edgelist_gdf["src"], edgelist_gdf["dst"]),
-                 },
-                "view_adj_list":
-                {"obj": G,
-                 },
-                "degree":
-                {"obj": G,
-                 },
-                "degrees":
-                {"obj": G,
-                 },
-                }
-    return algoData
-
-
-def loadDataFile(file_name, file_type, delimiter=' '):
+def getBenchmarks(G, edgelist_gdf, args):
+    benches = [
+        Benchmark(name="pagerank",
```

Comment:
So, when adding a new feature in cuGraph, we just need to :
1. add it in `getBenchmarks` in order to see it in the benchmark runs
2. update `parseCLI` with relevant parameters specific to the new feature

Reply:
Correct? I think I'll add that to the checklist we should go through when adding a new algo.

Reply:
yes, that's correct